addons/im_livechat/static/src/models/thread/thread.js

```diff
@@ -1,10 +1,34 @@
 /** @odoo-module **/
 
-import { patchModelMethods, patchRecordMethods } from '@mail/model/model_core';
-import { insert, link, unlink } from '@mail/model/model_field_command';
+import { addFields, addRecordMethods, patchModelMethods, patchRecordMethods } from '@mail/model/model_core';
+import { many2one } from '@mail/model/model_field';
+import { clear, insert, link, unlink } from '@mail/model/model_field_command';
 // ensure that the model definition is loaded before the patch
 import '@mail/models/thread/thread';
 
+addFields('mail.thread', {
+    /**
+     * If set, current thread is a livechat.
+     */
```

Comment:
not exactly correct, as if `this.messaging.discuss` is not set, this will not be set, but thread will still be a livechat

Reply:
Handling of unset `this.messaging.discuss` happens due to how rudimentary internal code of models work.
It's not desirable to handle it, it's just we have to live with it for the time being.

IMO it's conceptually correct.

Reply:
discuss should become not set on the guest page, it's a legitimate use case to support

Reply:
changed target to `mail.messaging` instead of `mail.discuss`